modulo5/projeto-cookenu/cookenu-back/src/Controller/UserController.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from "express";
+import { UserBusiness } from "../Business/UserBusiness";
+import { LoginUserInputDTO, UserInputDTO } from "../Model/Types/User";
+
+export class UserController {
+  private userBusiness: UserBusiness;
+  constructor() {
+    this.userBusiness = new UserBusiness();
+  }
+  public signUp = async (req: Request, res: Response) => {
+    try {
+      const input: UserInputDTO = {
+        name: req.body.name,
+        email: req.body.email,
+        password: req.body.password,
+      };
+      const token = await this.userBusiness.createUser(input);
+      res
+        .status(201)
+        .send({ message: `User ${input.name} was created!`, token });
+    } catch (error: any) {
+      res.status(400).send(error.message);
```

Comment:
Aqui, você setando o erro 400, todo e qualquer erro que você colocar vai acabar vindo pra cá pro catch. Nesse sentido, da uma lida sobre o instance Of para ver os erros aqui!